src/pairs/GameState.py

```diff
@@ -0,0 +1,34 @@
+from src.pairs.Player import Player
+from src.pairs.util import intMapper
+from collections import deque, Iterable
+
+class GameState:
+    def __init__(self, deck, target_score, players):
+        self.deck = deck
+        self.target_score = target_score
+        if isinstance(players, int):
+            self.num_players = players
+            self.players = deque(Player(deck) for _ in range(self.num_players))
+        elif isinstance(players, Iterable):
+            self.players = deque(players)
+            self.num_players = len(self.players)
+
+    def copy(self):
+        c = GameState(self.deck, self.target_score, self.players)
```

Comment:
Try renaming this to `deepcopy` instead of `copy` and make sure to copy the players instead of passing the same deque of players in.

Reply:
done